unit-tests/src/test/java/org/eclipse/collections/impl/collector/Collectors2Test.java

```diff
@@ -1015,4 +1018,154 @@ public void toImmutableBagMultimap2Parallel()
                 LARGE_INTERVAL.reduceInPlace(Collectors2.toImmutableBagMultimap(Object::toString, Object::toString)),
                 this.bigData.parallelStream().collect(Collectors2.toImmutableBagMultimap(Object::toString, Object::toString)));
     }
+
+    @Test
+    public void chunk()
+    {
+        MutableList<MutableList<Integer>> chunked = this.bigData.stream().collect(Collectors2.chunk(100));
```

Comment:
Tests where the collection size is not divisible by batch size are absent.

Please add a test where size%batch != 0 and size < batch, size == batch.